src/elements/content-sidebar/activity-feed/comment/BaseComment.js

```diff
@@ -0,0 +1,302 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import { FormattedMessage } from 'react-intl';
+import TetherComponent from 'react-tether';
+import Checkmark16 from '../../../../icon/line/Checkmark16';
+import Trash16 from '../../../../icon/line/Trash16';
+import Pencil16 from '../../../../icon/line/Pencil16';
+import X16 from '../../../../icon/fill/X16';
+import Avatar from '../Avatar';
+import Media from '../../../../components/media';
+import { MenuItem } from '../../../../components/menu';
+import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
+import DeleteConfirmation from '../common/delete-confirmation';
+import ActivityTimestamp from '../common/activity-timestamp';
+import UserLink from '../common/user-link';
+import ActivityCard from '../ActivityCard';
+import ActivityError from '../common/activity-error';
+import ActivityMessage from '../common/activity-message';
+import ActivityStatus from '../common/activity-status';
+import CommentForm from '../comment-form';
+import { COMMENT_STATUS_OPEN, COMMENT_STATUS_RESOLVED, PLACEHOLDER_USER } from '../../../../constants';
+import messages from './messages';
+import type { GetAvatarUrlCallback, GetProfileUrlCallback } from '../../../common/flowTypes';
+import type { Translations } from '../../flowTypes';
+import type { SelectorItems, User } from '../../../../common/types/core';
+import type { ActionItemError, BoxCommentPermission, FeedItemStatus } from '../../../../common/types/feed';
+import './Comment.scss';
+
+type Props = {
+    created_at: string | number,
+    created_by: User,
+    currentUser?: User,
+    error?: ActionItemError,
+    getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
+    id: string,
+    isDisabled?: boolean,
+    isPending?: boolean,
+    mentionSelectorContacts?: SelectorItems<>,
+    modified_at?: string | number,
+    onDelete: ({ id: string, permissions?: BoxCommentPermission }) => any,
+    onEdit: (
+        id: string,
+        text?: string,
+        status?: FeedItemStatus,
+        hasMention: boolean,
+        permissions: BoxCommentPermission,
+        onSuccess: ?Function,
+        onError: ?Function,
+    ) => void,
+    onSelect: (isSelected: boolean) => void,
+    permissions: BoxCommentPermission,
+    status?: FeedItemStatus,
+    tagged_message: string,
+    translatedTaggedMessage?: string,
+    translations?: Translations,
+};
+
+// TODO: Replace and rename to Comment component once threaded replies refactor is fully implemented
```

Comment:
what's the reason for both `BaseComment` and `Comment`? why not refactor the `Comment` file directly?

Reply:
We are putting the threaded replies refactor behind a ff

Reply:
does this file(`BaseComment.js`) anticipate additional changes after this PR?

if not, it's probably best to not have this particular refactor behind a ff since it's straight forward with no change to logic. this way there's less clean up once the ff is released and overall less code being released all at one time.

Reply:
It will have additional changes to `BaseComment.js`